### DISCUSSION TITLE
CMS Contentful - Example doesn't deploy with post entries

### DISCUSSION THREAD

**Discussion by Harrison-Frost** [vercel/next.js#20880] [Help]
When I view my local preview the posts are connected to the CMS. But now i've deployed it with vercel, the live URL doesn't display any post entries. Why is this?

**Comment by doodirock** [comment]
did you ever figure this out?  Its broken for me in both cases.  Cant get it to connect

**Reply by leerob** [reply]
Did you add the correct environment variables needed to Vercel?

https://vercel.com/docs/environment-variables

**Reply by doodirock** [reply]
Looks like the ones that vercel adds automatically actually doesn't work. I had to regenerate new keys

**Reply by leerob** [reply]
According to the `cms-contentful` example, it sets these keys:

```
// .env.local
CONTENTFUL_SPACE_ID=
CONTENTFUL_ACCESS_TOKEN=
CONTENTFUL_PREVIEW_ACCESS_TOKEN=
CONTENTFUL_PREVIEW_SECRET=
```

which are the same keys set by the [Vercel Contentful integration](https://vercel.com/integrations/contentful).

<img width="765" alt="CleanShot 2021-08-09 at 22 44 25@2x" src="https://user-images.githubusercontent.com/9113740/128805123-14248422-3ae4-4650-8098-6a5c9cc55c4b.png">

**Reply by andrewkess** [reply]
> Looks like the ones that vercel adds automatically actually doesn't work. I had to regenerate new keys

Thank you, this worked for me. I literally bootstrapped this project twice and spent a few hours trying to figure out why none of my posts displayed, even though I used the same API keys as set up by the Vercel Contentful integration. It wasn't until I just ignored the keys that were automatically generated by the "Deploy now" button and set my own keys that it finally started working. It's a shame that the one-click deployment option is most likely tripping up others too...
